varcode/variant_collection.py

```diff
@@ -132,9 +138,9 @@ def clone_with_new_elements(self, new_elements):
         """
         return self.__class__(
             new_elements,
-            path=self.path,
+            source_to_metadata_dict=self.source_to_metadata_dict,
```

Comment:
Yeah, I think the difference seemed reasonable when I was considering having `Collection` only take `sources` but not `source_to_metadata_dict`.